### PR TITLE
[SPARK-38304][SQL] Elt() should return null if index is null under ANSI mode

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Elt() should return null if the input index is null under ANSI mode, which is consistent with MySQL where the function is from.
Before changes:
image

After changes:
The query returns null.  

### Why are the changes needed?

Bug fix

### Does this PR introduce _any_ user-facing change?

Yes, SQL function Elt() returns null if the input index is null under ANSI mode, instead of runtime error.

### How was this patch tested?

UT

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala [hunk 1]
```diff
@@ -361,15 +361,19 @@ case class Elt(
...
-         |$inputVal = null;
-         |do {
-         |  $codes
-         |} while (false);
-         |$indexOutOfBoundBranch
-         |final ${CodeGenerator.javaType(dataType)} ${ev.value} = $inputVal;
-         |final boolean ${ev.isNull} = ${ev.value} == null;
+         |boolean ${ev.isNull} = ${index.isNull};
+         |${CodeGenerator.javaType(dataType)} ${ev.value} = null;
+         |if (!${index.isNull}) {
```
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala:366]: If the index is null, skip the following loop.

**Comment by gengliangwang** [comment]
cc @cloud-fan @anchovYu

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master/3.2!

**Comment by anchovYu** [comment]
thanks!

**Pull request closed without merging**